src/atoms/Text/index.js

```diff
@@ -0,0 +1,82 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import colors from 'atoms/Color/colors.module.scss';
+import styles from './text.module.scss';
+
+function Text(props) {
+  if (!props.children) return null;
+
+  const {
+    children,
+    className,
+    tag,
+    type,
+    weight,
+    color,
+    variant,
+  } = props;
+
+  const Element = `${tag}`;
+  const classes = [
+    styles[`typography-${type}`],
+    styles[weight],
+    colors[color],
+    styles[variant],
+    className
+  ];
+
+  return (
+    <Element className={classnames(...classes)}>
+      { children }
+    </Element>
+  );
+}
+
+
+Text.propTypes = {
+  /**
+   * This prop will add a new className to any inherent classNames
+   * provided in the component's index.js file.
+   */
+  className: PropTypes.string,
+  /**
+   * You can use any html element text tag
+   */
+  tag: PropTypes.string,
+
+  /**
+   * determines typography class
+   */
+  type: PropTypes.number,
+
+  /**
+   * Font weight: light|regular|bold
+   * - light = 300
+   * - semibold = 500
+   * - bold = 600
```

Comment:
This is really confusing. My understanding is that we only have light, regular, and semibold options. `regular` should probably be the default.